File: backend/src/main/java/com/plugga/backend/dao/PileDAO.java
package com.plugga.backend.dao;

import com.plugga.backend.entity.Pile;

import java.util.List;

public interface PileDAO {
    List<Pile> findAll();

    Pile findById(int id);

    void save(Pile pile);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/dao/PileDAO.kt
package com.plugga.backend.dao

import com.plugga.backend.entity.Pile
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface PileDAO : PagingAndSortingRepository<Pile, Int> {

    fun findByDeckId(deckId: Int): List<Pile>

    fun findByDeckId(pageable: Pageable, deckId: Int): Page<Pile>
}
